51.md

```diff
@@ -6,148 +6,116 @@ Lists
 
-```json
-{
-  "kind": 10000,
-  "tags": [
-    ["p", "3bf0c63fcb93463407af97a5e5ee64fa883d107ef9e558472c4eb9aaaefa459d"],
-    ["p", "32e1827635450ebb3c5a7d12c1f8e7b2b514439ac10a67eef3d9fd9c5c68e245"],
-  ],
-  "content": "VezuSvWak++ASjFMRqBPWS3mK5pZ0vRLL325iuIL4S+r8n9z+DuMau5vMElz1tGC/UqCDmbzE2kwplafaFo/FnIZMdEj4pdxgptyBV1ifZpH3TEF6OMjEtqbYRRqnxgIXsuOSXaerWgpi0pm+raHQPseoELQI/SZ1cvtFqEUCXdXpa5AYaSd+quEuthAEw7V1jP+5TDRCEC8jiLosBVhCtaPpLcrm8HydMYJ2XB6Ixs=?iv=/rtV49RFm0XyFEwG62Eo9A==",
-  ...other fields
-}
-```
+## Sets
 
+Sets are lists with well-defined meaning that can enhance the functionality and the UI of clients that rely
+on them. Unlike standard lists, users are expected to have more than one set of each kind, therefore each of
+them must be assigned a different `"d"` identifier.
 
-## Parameterized Replaceable List Event Example
+For example, _relay sets_ can be displayed in a dropdown UI to give users the option to switch to which
+relays they will publish an event or from which relays they will read the replies to an event; _curation sets_
+can be used by apps to showcase curations made by others tagged to different topics.
 
-Lets say a user wants to create a 'Categorized People' list of `nostr` people and has keys:
-```
-priv: fb505c65d4df950f5d28c9e4d285ee12ffaf315deef1fc24e3c7cd1e7e35f2b1
-pub: b1a5c93edcc8d586566fde53a20bdb50049a97b15483cb763854e57016e0fa3d
-```
-The user wants to publicly include these users:
+Aside from their main identifier, the `"d"` tag, sets can optionally have a `"name"`, a `"picture"` and an
+`"about"` tag that can be used to enhance their UI.
 
-```json
-["p", "3bf0c63fcb93463407af97a5e5ee64fa883d107ef9e558472c4eb9aaaefa459d"],
-["p", "32e1827635450ebb3c5a7d12c1f8e7b2b514439ac10a67eef3d9fd9c5c68e245"]
-```
-and privately include these users (below is the JSON that would be encrypted and placed in the event content):
+| name          | kind  | description                                                                                   | expected tag items                                |
+| ---           | ---   | ---                                                                                           | ---                                               |
+| Follow sets   | 30000 | categorized groups of users a client may choose to check out in different circumstances       | `"p"` (pubkeys)                                   |
+| Relay sets    | 30002 | user-defined relay groups the user can easily pick and choose from during variadic operations | `"relay"` (relay URLs)                            |
+| Curation sets | 30004 | groups of articles picked by users as interesting and/or belonging to the same category       | `"a"` (kind:30023 articles), `"e"` (kind:1 notes) |
+| Emoji sets    | 30030 | categorized emoji groups                                                                      | `"emoji"` (see [NIP-30](30.md))                   |
```

Comment:
Suggestion to add: `Interest Set` which is kind `30015`, and takes a set of `t` tags.

Reply:
@pablof7z is it worth also mentioning Hightlight lists?

Reply:
> We're doing something similar in YakiHonne related to the user preferences for content customizations via `kind-30100`. I think its better to include such specifications once for all because soon or later feeds customization will be faced (at least thats what we experienced).
> 
> ![content_settings](https://user-images.githubusercontent.com/29741039/283314136-db1b35f3-a884-422b-8252-8f0ca353d85d.jpg) ![topics](https://user-images.githubusercontent.com/29741039/283314148-cd974bee-589f-4d0e-b40a-65a9290447bc.jpg)
> 
> ```
> {
>   "id": "ed4291ff8ba4343ca444d03624afcee98257aba4af17410a73144ffdc369c467",
>   "kind": 30100,
>   "pubkey": "1cb0ebecfbe8e291d46c4e082df1e6a183bb321245ef29df5e2703f84b1d86e1",
>   "created_at": 1700102857,
>   "content": "",
>   "tags": [
>     [
>       "d",
>       "MyFavoriteTopicsInYakihonne"
>     ],
>     [
>       "t",
>       "YakiHonne"
>     ],
>     [
>       "t",
>       "Technology"
>     ],
>     [
>       "t",
>       "Crypto"
>     ],
>     [
>       "t",
>       "Travel"
>     ],
>     [
>       "t",
>       "Social"
>     ],
>     [
>       "t",
>       "Nostr"
>     ],
>     [
>       "t",
>       "Economy"
>     ],
>     [
>       "t",
>       "Finance"
>     ],
>     [
>       "t",
>       "Entrepreneurship"
>     ],
>     [
>       "t",
>       "Self improvement"
>     ],
>     [
>       "t",
>       "Food"
>     ],
>     [
>       "t",
>       "Marketing"
>     ],
>     [
>       "t",
>       "Art"
>     ],
>     [
>       "t",
>       "Photography"
>     ],
>     [
>       "t",
>       "Entertainment"
>     ],
>     [
>       "t",
>       "Music"
>     ]
>   ],
>   "sig": "4fdc6a423bc7320b432e2edc1e794895468894b92ad7bb9e119bd16c1a6c1d0098bbd8a33ac25856acbae222fda565d7407c17d2b09f6b326de3762ad30a8a1c"
> }
> ```

How about this from above it complies in your side and some of what @staab wants to achieve I Think his proposition is based on the advance search he has on coracle.social

Reply:
@Darecjo if your list is automatically named `MyFavoriteTopicsInYakihonne` I think that is an indication that you should be using NIP-78. What do you think?

@erskingardner would these things be a single standard list or multiple? It's hard to imagine someone having _interests_, which is such a generic concept, categorized in different groups. I feel like each interest is already a "group" on its own.

On the other hand one interest is translated into potentially huge number of tags -- i.e. if I'm interested in "fruits" I may have the hashtags "banana", "mango", "passionfruit" etc. Or if I'm interested in "banana" I may have "banana", "cavendish", "bananas", "plaintain", "banana-recipes", "cooking-bananas" whatever.

So what do we do? Add both a canonical standard interests list and interest sets?

Reply:
Good question. @pablof7z added `30015` to NDK a while back and we've both been using it as a PRE. I think that there is a good reason to keep separate lists, e.g. I might have a music interests lists with music related stuff, and also an outdoors list that is all about climbing and skiing. I might want to use one or the other to view curated topic feeds.

Maybe that means there is a case to do both `10015` and `30015`, in similar style to emoji lists. Where your `10015` list is the full canonical list and can include either `t` tags or `a` tags pointing to other `30015` lists.

Reply:
> @Darecjo if your list is automatically named `MyFavoriteTopicsInYakihonne` I think that is an indication that you should be using NIP-78. What do you think?

We are going to check the interests list and sets since they are more suitable for our case it is basically the same thing.